### PR TITLE
Decouple the association between events and scans

### PR DESCRIPTION
**Pull request by jedschneider**
since events will be created and destroyed as the schedule changes, we need to _not_ have a relationship between the two, but still need to query in order to link the two.

### CODE REVIEW

**Comment by jedschneider** [comment]
@NickyBobby for your review. Anything I missed?

File: app/models/event.rb [hunk 1]
```diff
@@ -3,9 +3,12 @@ class Event < ActiveRecord::Base
...
   belongs_to :creator, class_name: "Teacher"
-  has_many :scans, dependent: :destroy
 
   def scanned_in?
-    !scans.where(correct: true).empty?
+    scans.where('timestamp > ? AND timestamp < ?', start_time, end_time).present?
+  end
+
+  def scans
+    Scan.where(correct: true, student:student, location: location)
```
Comment (NickyBobby) [inline app/models/event.rb:12]: So I see what's happening here and it looks good, but what is the method that changes the correct boolean attribute on scans to true or false? I looked on the Student and Scan models and I don't see it there. I'm sure I'm just not looking in the right spot. Also I noticed that scans still have an event_id attribute on the model. We want to remove that as well, correct? @jedschneider

File: app/models/scan.rb [hunk 2]
```diff
@@ -1,5 +1,4 @@
 class Scan < ActiveRecord::Base
-  belongs_to :event
```
Comment (NickyBobby) [inline app/models/scan.rb:2]: Ok so scans no longer belong to an event but do we still need the event_id attribute on the Scan model?

File: app/models/event.rb (hunk 1 above)
Comment (jedschneider) [inline app/models/event.rb:12]: as for scans, correct or not, that is done when the [scan is created](https://github.com/rootselementary/check_in_tool_rails/blob/2cb38520f45dd8cac0a6d3abd582f9f674176e4a/app/models/scan_creator.rb#L7). Good catch on fixing up the schema. I'll add that to the migration.

**Pull request merged**